course05/homework01/mousamroy/string.cc

```diff
@@ -0,0 +1,58 @@
+#include "string.h"
+
+String::String()
+{
+	data[0] = 0;
+}
+
+String::String(const std::string& str)
+{
+	if (str.size() > maxLength)
+	{
+		throw std::invalid_argument("max string length exceeded");
+	}
+
+	std::size_t i;
+	for (i = 0; i < str.size(); ++i)
+	{
+		data[i] = str[i];
+	}
+
+	data[i] = 0;
+}
+
+bool String::Empty() const
+{
+	return Size() == 0;
+}
+
+std::size_t String::Size() const
+{
+	for (std::size_t i = 0; i <= maxLength; ++i)
+	{
+		if (data[i] == 0)
+			return i;
+	}
```

Comment:
This linear complexity is a bit unexpected and can lead to slowness, as `Size()` is very often use in loops. You should consider adding `size_type mSize` as attribute